components/layout_2020/style_ext.rs

```diff
@@ -494,3 +494,29 @@ fn size_to_length(size: &Size) -> LengthPercentageOrAuto {
         Size::Auto => LengthPercentageOrAuto::Auto,
     }
 }
+
+pub(crate) fn transferred_inline_min_max(
+    ratio: f32,
+    style: &ComputedValues,
+    containing_block: &ContainingBlock,
+    pbm: &PaddingBorderMargin,
+) -> (Length, Option<Length>) {
+    let max_block_size = style
+        .max_box_size(containing_block.style.writing_mode)
+        .block
+        .map(|l| l.maybe_percentage_relative_to(containing_block.block_size.non_auto()))
+        .flatten();
+    let min_block_size = style
+        .min_box_size(containing_block.style.writing_mode)
+        .block
+        .maybe_percentage_relative_to(containing_block.block_size.non_auto())
+        .auto_is(|| Length::zero());
+
+    match style.get_position().box_sizing {
+        BoxSizing::ContentBox => (min_block_size * ratio, max_block_size.map(|b| b * ratio)),
+        BoxSizing::BorderBox => (
+            min_block_size * ratio - pbm.padding_border_sums.inline,
+            max_block_size.map(|b| b * ratio - pbm.padding_border_sums.inline),
```

Comment:
It the  ratio refers to the inner size, shouldn't this be more like

```rs
(min_block_size - pbm.padding_border_sums.block) * ratio,
max_block_size.map(|b| (b - pbm.padding_border_sums.block) * ratio),
```

Reply:
imo max_block_size comes from style so that's what should be transferred, right?

Reply:
I'm checking

```html
<!DOCTYPE html>
<div style="display: flex">
  <img src="image" style="box-sizing: border-box; min-height:300px; border: solid 100px">
</div>
<script>
addEventListener("load", () => document.body.append(document.querySelector("img").getBoundingClientRect().width);
</script>
```

where image is 200px wide and 100px tall (ratio of 2).

Your code takes `min-height:300px`, multiplies it by 2, and subtracts the horizontal border. So the inner width is `300*2 - 200 = 400`, and the outer width is 600.

But you can see that in other browsers, the inner width is 200, and the outer width is 400. That's because the inner height is 100, so they preserve the content ratio of 2.